package.json

```diff
@@ -129,6 +129,7 @@
 		"test": "npm run lint && npm run test-unit",
 		"test-php": "npm run lint-php && npm run test-unit-php",
 		"ci": "concurrently \"npm run lint && npm run build\" \"npm run test-unit:coverage-ci\"",
+		"license-check": "! npm ls --production --parseable | xargs -I {} jq --raw-output '.name + \" \" + ( .license // .licenses[0].type )' '{}/package.json' | grep -v -E '^.* .*(MIT|GPL-2|ISC|BSD|CC0).*$'",
```

Comment:
npm scripts are usually meant to be portable. My gut feeling is that this command is way too 🔥 for Windows.

Since we only care about Travis CI catching these errors, maybe move it over into e.g. `./bin/check-license.sh`?

Reply:
Having said that, it looks like most of our npm scripts aren't portable... 😅

Reply:
Pretty much every part of this command will fail on Windows. 🙃

I think it's a good idea to put it in a script: makes it easier to format the output, so there's extra information about why it failed.

Reply:
>Since we only care about Travis CI catching these errors, maybe move it over into e.g. `./bin/check-license.sh`?

Or alternatively `./bin/check-license.js` where we could use cross-platform Node modules.

There's a few: https://www.npmjs.com/search?q=license%20check

I also swear I've seen some build integrations which support this as a feature, though I'm coming up short on links.

Reply:
Aha, here it is: https://fossa.io/

Reply:
"DETECTS HIDDEN GPL" is probably my favourite marketing line on that site. 😂

Reply:
Yes, we should use more Node for those scripts to make them cross-platform.